bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandlerFactory.java

```diff
@@ -128,9 +139,71 @@ public ThingHandler registerHandler(Thing thing) {
         setHandlerContext(thingHandler);
         registerConfigStatusProvider(thing, thingHandler);
         registerFirmwareUpdateHandler(thing, thingHandler);
+        registerServices(thing, thingHandler);
         return thingHandler;
     }
 
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private void registerServices(Thing thing, ThingHandler thingHandler) {
+        ThingUID thingUID = thing.getUID();
+        for (Class c : thingHandler.getServices()) {
+            Object serviceInstance;
+            try {
+                serviceInstance = c.newInstance();
+
+                if (serviceInstance instanceof ThingHandlerService) {
+                    ThingHandlerService ths = (ThingHandlerService) serviceInstance;
+                    ths.setThingHandler(thingHandler);
+                } else {
+                    logger.warn(
+                            "Should register service={} for thingUID={}, but it does not implement the interface ThingHandlerService.",
+                            c.getCanonicalName(), thingUID);
+                    continue;
+                }
+
+                Dictionary<String, String> properties = new Hashtable<>();
+                properties.put(ThingHandlerService.ACTION_THING_UID, thingUID.getAsString());
```

Comment:
Maybe calculating the properties should be the responsibility of the handler? We might define just another method on the ThingHandler `Dictionary<String, String> getThingHandlerServiceProperties(ThingHandlerService ths)` and the ThingHandler can decide which properties are necessary. Otherwise every THS would have the `ACTION_THING_UID` property but no use for it.